multipers/gudhi/truc.h

```diff
@@ -628,25 +629,22 @@ class Truc {
     return out;
   }
 
-  Truc<PersBackend, Structure, typename MultiFiltration::template Base<int32_t>>
-  coarsen_on_grid(
-      const std::vector<std::vector<typename MultiFiltration::value_type>>
-          grid) {
-    std::vector<typename MultiFiltration::template Base<int32_t>> coords(
-        this->num_generators());
+  auto coarsen_on_grid(
```

Comment:
`auto` return type ??

Reply:
Yes. Makes the problem easier in this particular case.